src/rtodms.cpp

```diff
@@ -45,10 +45,19 @@ rtodms(char *s, size_t sizeof_s, double r, int pos, int neg) {
 	int deg, min, sign;
 	char *ss = s;
 	double sec;
+	size_t sizeof_ss = sizeof_s;
 
 	if (r < 0) {
 		r = -r;
-		if  (!pos) { *ss++ = '-'; sign = 0; }
+		if  (!pos) {
+		    if( sizeof_s == 1 ) {
+		        *s = 0;
+		        return s;
+		    }
+		    sizeof_s --;
```

Comment:
`sizeof_ss--` :)

Reply:
sigh... thanks again. commit amended with the fix and more extensive tests, to be faithful with my motto "untested code is broken code"

Reply:
LGTM now. thanks!